### PR TITLE
Include order_detail information in Shipment.allocations

### PR DESCRIPTION
**Pull request by miggland**
Fix for issue #3382 

<a href="https://gitpod.io/#https://github.com/inventree/InvenTree/pull/3383"></a>

### CODE REVIEW

File: InvenTree/order/serializers.py [hunk 1]
```diff
@@ -912,7 +912,7 @@ class Meta:
 class SalesOrderShipmentSerializer(InvenTreeModelSerializer):
     """Serializer for the SalesOrderShipment class."""
 
-    allocations = SalesOrderAllocationSerializer(many=True, read_only=True, location_detail=True)
+    allocations = SalesOrderAllocationSerializer(many=True, read_only=True, location_detail=True, order_detail=True)
```
Comment (matmair) [inline InvenTree/order/serializers.py:915]: Suggestion: maybe make both location and order detail a flag to reduce unneeded traffic.

**Review by matmair — APPROVED** [review]

**Comment by matmair** [comment]
@miggland sample for the suggested switches:
https://github.com/inventree/InvenTree/blob/96faa0d5f053bccfd005059c6e7ca89d0785d42a/InvenTree/order/serializers.py#L91-L98

**Comment by SchrodingersGat** [comment]
@miggland I'm a bit confused here, isn't the detail you want already available at that API endpoint?

image

**Comment by miggland** [comment]
You're right @SchrodingersGat , I obviously didn't check the result properly..

That's not the information I wanted. What I want, as described in #3382, is actually the corresponding order line information.

**Comment by miggland** [comment]
This is what I wanted:
image

Adding the `SalesOrderLineItemMinimalSerializer` feels a bit hacky, but I don't directly see a better way of doing this (since `SalesOrderLineItemSerializer` depends on `SalesOrderAllocationSerializer`, it has to be defined afterwards, and using this method would lead to a loop..

**Comment by SchrodingersGat** [comment]
You're correct about avoiding a loop. TBH this use case seems (to me) to be bordering on a situation where simply requesting the lines for a given order (only one extra API call) is sufficient. Can you describe your desired use case here?

**Comment by miggland** [comment]
The background is that I was starting on creating packing lists - ie reports of what is in a single shipment. I want the report to include the same information as the order confirmation, but be specific as to what is in a certain shipment.

While working on this I realised the problem - but thought I'd finish it up with a possible solution anyway. I'm not convinced it's a good idea either.

The alternative is to ask for each line item specifically while looping through a shipment - not hard, but it's either more than one API call or a search in the results returned. I think it's possible to put this in the python interface instead, as a method of the Shipment class 'getOrderLines' or whatever.

**Comment by SchrodingersGat** [comment]
@miggland I think given the (almost) recursion here, and the duplication of existing data, I'd prefer not to merge this one in. Having multiple ways to achieve the same thing via the API gets confusing and hard to maintain :)

**Comment by miggland** [comment]
Fair enough, I agree.

**Pull request closed without merging**